app/Http/Controllers/Admin/ProjectController.php

```diff
@@ -0,0 +1,179 @@
+use Illuminate\Http\Response;
+use Illuminate\Support\Str;
+
+/**
+ * Class ProjectController
+ *
+ * @package App\Http\Controllers\Admin
+ */
+class ProjectController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function index()
+    {
+        return view('admin.pages.projects.index', ['projects' => Project::all()]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function create()
+    {
+        return view('admin.pages.projects.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  ProjectRequest $projectRequest
+     *
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function store(ProjectRequest $projectRequest)
+    {
+        $project = new Project();
+
+        $project->name = $projectRequest->name;
+        $project->description = $projectRequest->description;
+        $project->url = $projectRequest->url;
+        $project->image = $this->uploadImage($projectRequest);
+        $project->slug = Str::slug($projectRequest->name);
+        $project->save();
+
+        return redirect()->route('project.index')->with('status', 'Successfully Inserted!');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  Project  $project
+     *
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function show(Project $project)
+    {
+        return view('admin.pages.projects.show', ['project' => $project]);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  Project  $project
+     *
+     * @return Response
+     */
+    public function edit(Project $project)
+    {
+        return view('admin.pages.projects.edit', ['project' => $project]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  ProjectRequest $projectRequest
+     * @param  Project  $project
+     *
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function update(ProjectRequest $projectRequest, Project $project)
+    {
+        $project->name = $projectRequest->name;
+        $project->description = $projectRequest->description;
+        $project->url = $projectRequest->url;
+        $project->image = $this->updateImage($projectRequest, $project->image);
+        $project->slug = Str::slug($projectRequest->name);
+        $project->save();
+
+        return redirect()->route('project.index')->with('status', 'Successfully Updated!');
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  Project  $project
+     *
+     * @throws \Exception
+     * @author Ali, Muamar
+     *
+     * @return Response
+     */
+    public function destroy(Project $project)
```

Comment:
That is the default controller method names because in Laravel they have a naming convention in controller methods names.
"For the controller methods the conventional names are index, show, create, store, edit, update and delete. And snakeCase for custom methods."

Reply:
noted